### PR TITLE
aaf: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

File: recipes/aaf/all/conanfile.py [hunk 1]
```diff
@@ -41,50 +44,49 @@ def validate(self):
...
-    def build(self):
-        apply_conandata_patches(self)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-        cmake = CMake(self)
-
-        if tools.is_apple_os(self.settings.os):
-            cmake.definitions["PLATFORM"] = "apple-clang"
+    def generate(self):
+        tc = CMakeToolchain(self)
```
Comment (MartinDelille) [inline recipes/aaf/all/conanfile.py:50]:
```suggestion
        tc = CMakeToolchain(self)
        arch = "x86_64"
```

File: recipes/aaf/all/conanfile.py [hunk 2]
```diff
@@ -41,50 +44,49 @@ def validate(self):
...
+        if is_apple_os(self):
+            tc.cache_variables["PLATFORM"] = "apple-clang"
```
Comment (MartinDelille) [inline recipes/aaf/all/conanfile.py:52]:
```suggestion
            tc.cache_variables["PLATFORM"] = "apple-clang"
            if self.settings.arch == "armv8":
                arch = "arm64"
```

File: recipes/aaf/all/conanfile.py [hunk 3]
```diff
@@ -41,50 +44,49 @@ def validate(self):
...
         elif self.settings.compiler == "Visual Studio":
-            cmake.definitions["PLATFORM"] = "vc"
+            tc.cache_variables["PLATFORM"] = "vc"
```
Comment (MartinDelille) [inline recipes/aaf/all/conanfile.py:54]:
```suggestion
            tc.cache_variables["PLATFORM"] = "vc"
            arch = "x64"
```

File: recipes/aaf/all/conanfile.py [hunk 4]
```diff
@@ -41,50 +44,49 @@ def validate(self):
...
         else:
-            cmake.definitions["PLATFORM"] = self.settings.os
+            tc.cache_variables["PLATFORM"] = str(self.settings.os)
+        tc.cache_variables["ARCH"] = "x86_64"  # ARCH is used only for setting the output directory. So itsvalue does not matter here.
```
Comment (MartinDelille) [inline recipes/aaf/all/conanfile.py:57]:
```suggestion
        tc.cache_variables["ARCH"] = arch
```

File: recipes/aaf/all/conanfile.py [hunk 5]
```diff
@@ -41,50 +44,49 @@ def validate(self):
             raise ConanInvalidConfiguration("ARM v8 not supported")
```
Comment (MartinDelille) [inline recipes/aaf/all/conanfile.py:44]: Apple Silicon is supported since 1.2.0 RC1: https://sourceforge.net/p/aaf/code2/ci/master/tree/

**Review by MartinDelille — CHANGES_REQUESTED** [review]

**Comment by SpaceIm** [comment]
@MartinDelille it's out of scope of this PR, but I've added macOS M1 support.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Pull request merged**